.github/workflows/deploy-standalone-plugins.yml

```diff
@@ -12,88 +13,81 @@ on:
-      - name: Get plugin version
-        id: get-version
-        if: ${{ github.event_name == 'workflow_dispatch' }}
-        run: |
-          echo "version=$(node ./bin/plugin/cli.js get-plugin-version --slug=${{ inputs.slug }})" >> $GITHUB_OUTPUT
+
       - name: Set matrix
         id: set-matrix
         run: |
+          PLUGINS=$(jq -r '.plugins' plugins.json)
           if ${{ github.event_name == 'workflow_dispatch' }}; then
-            result=$(echo "${{ steps.get-version.outputs.version }}" | awk '/^(\*|[0-9]+(\.[0-9]+){0,2}(-[a-zA-Z0-9.]+)?)$/ {print "Matched"}')
-            if [[ -n "$result" ]]; then
-              # Set the manual input values in JSON format for use in the matrix.
-              echo "matrix={\"include\":[{\"slug\":\"${{ inputs.slug }}\",\"version\":\"${{ steps.get-version.outputs.version }}\",\"directory\":\"${{ steps.get-plugin-directory.outputs.directory }}\",\"dry-run\":\"${{ inputs.dry-run }}\"}]}" >> $GITHUB_OUTPUT
-            else
-              echo "The ${{ inputs.slug }} module slug is missing in the file plugins.json."
+            SLUG=${{ inputs.slug }}
+
+            # Check if the slug is valid
+            if [[ ! " $PLUGINS " =~ " $SLUG " ]]; then
+              echo "Invalid slug: $SLUG"
               exit 1
             fi
-          else
-            # Load the JSON file and parse from "{name: {slug, version}, ...}" to "include: [{ name, slug, version }, ...]"
-            # for use in the matrix.
-            # For plugins, the "version" parameter is not included here; it will dynamically get it in its own job.
 
-            # The "dry-run" parameter is included here to set the deployment mode.
-            # When running the manual (workflow_dispatch) workflow, this value will be set from manual input type.
-            echo "matrix=$(jq -c '{include: ([.modules | to_entries[] | {name:.key, slug: .value.slug, version: .value.version, directory: "build", "dry-run": false}] + ([.plugins[] | {name:. , slug:. , directory: "plugins", "dry-run": false}]))}' plugins.json)" >> $GITHUB_OUTPUT
+            PLUGINS="[ \"$SLUG\" ]"
           fi
+
+          # Set the matrix
+          echo "json=$(echo $PLUGINS | jq -c .)" >> $GITHUB_OUTPUT
+
   deploy:
     name: Deploy Plugin
-    needs: release
+    needs: pre-run
     runs-on: ubuntu-latest
     strategy:
-      matrix: ${{ fromJSON(needs.release.outputs.matrix) }}
+      matrix:
+        plugin: ${{ fromJSON(needs.pre-run.outputs.matrix) }}
     steps:
-      - name: Checkout repository
+      - name: Checkout
         uses: actions/checkout@v3
+
       - name: Setup Node.js (.nvmrc)
         uses: actions/setup-node@v3
         with:
           node-version-file: '.nvmrc'
           cache: npm
+
       - name: Install npm dependencies
-        run: npm ci
-      - name: Building standalone plugins
-        if: ${{ matrix.directory != 'plugins' }}
-        run: npm run build-plugins
-      - name: Set version
```

Comment:
@felixarntz @mukeshpanchal27 Can you please confirm if need to set this version? I think that was required for modules?

Reply:
This is needed to set the correct `VERSION` environment variable for the 10up deployment workflow. You removed this below, but it has to be maintained to make sure we publish the correct version.

Reply:
Thanks for confirming. I have re-added this. Now the Webpack generates the build manifest of plugin(s) which can provide version information.